### PR TITLE
refactoring utilities and udf modules

### PR DESCRIPTION
**Pull request by jamie256**
* Adding UDF namespace for language specific and component textstat metrics
* Ading consistent init to udf modules
* added two new composite modules to reduce the number of imports: llm_metrics and all_metrics
* split RollingLogger from openai wrapper, and moved to whylogs folder
* added some helper scripts that are whylogs specific to focus the content in the example notebooks
* Updated and fixed tests, including --load tests
* Moved optional UDF metrics dependencies to optional dependencies of langkit and add an all group for optional dependencies as a convenience.
* Minor updates to example notebooks and ran through each to verify, but this needs more work.

### CODE REVIEW

File: langkit/textstat.py [hunk 1]
```diff
@@ -29,58 +34,59 @@ def automated_readability_index(text: str) -> float:
...
 
 
-@register_metric_udf(col_type=String)
+@register_metric_udf(col_type=String, schema_name="text_standard_component")
 def gunning_fog(text: str) -> float:
     return textstat.textstat.gunning_fog(text)
 
 
 @register_metric_udf(col_type=String)
 def aggregate_reading_level(text: str) -> float:
```
Comment (FelipeAdachi) [inline langkit/textstat.py:53]: we don't need a schema_name here?

Comment (jamie256) [reply langkit/textstat.py:53]: I can put it back, but gunning_fog is part of the later text_standard overall readability metric, so I thought I'd remove the individual components?

Comment (FelipeAdachi) [reply langkit/textstat.py:53]: Sorry, I didn't understand the changes, I get it now, all good!

**Review by FelipeAdachi — APPROVED** [review]
LGTM!

**Pull request merged**